samza-core/src/main/java/org/apache/samza/coordinator/JobCoordinator.java

```diff
@@ -55,12 +59,16 @@
    * @throws InterruptedException if the current thread is interrupted while waiting for the JobCoordinator to start-up
    */
   boolean awaitStart(long timeoutMs) throws InterruptedException;
+
   /**
-   * Returns the logical ID assigned to the processor
-   * It is up to the user to ensure that different instances of StreamProcessor within a job have unique processor ID.
-   * @return integer representing the logical processor ID
+   * Returns the logical ID assigned to the processor that is local to the instance of StreamProcessor.
```

Comment:
nit: why is this called "logic"? I would think this is a "physical" ID since it is specific to the underlying environment already.